File: object_relations/app/models/restaurant.rb
# Restaurant has many customers
# Restaurant has many reviews through customers

class Restaurant
  attr_accessor :name
  @@all = []

  def initialize(name)
    @name = name
    @reviews = []
    @@all << self
  end

  def add_review(review)
    @reviews << review
  end

  def customers
    @reviews.collect do |review|
      review.customer
    end
  end

end

File: javascript/models/restaurant.js
class Restaurant {
  constructor(name) {
    this.name = name;
    this.reviews = [];
    Store.restaurants.push(this);
  }

  all() {
    return Store.restaurants;
  }

  findRestaurant(name) {
    return Store.restaurants.find(restaurant => { return restaurant.name === name; })
  }

  reviews() {

  }

  customers() {

  }

}
